src/rules/require-baseline.js

```diff
@@ -0,0 +1,564 @@
+		);
+	}
+
+	/**
+	 * Determines if any rule supports any property identifiers.
+	 * @param {string} property The name of the property.
+	 * @returns {boolean} `true` if any rule supports the property values, `false` if not.
+	 */
+	hasPropertyIdentifiers(property) {
+		return this.#rules.some(rule => rule.hasPropertyIdentifiers(property));
+	}
+
+	/**
+	 * Determines if any rule supports a function.
+	 * @param {string} property The name of the property.
+	 * @param {string} func The function to check.
+	 * @returns {boolean} `true` if any rule supports the function, `false` if not.
+	 */
+	hasPropertyFunction(property, func) {
+		return this.#rules.some(rule => rule.hasFunction(property, func));
+	}
+
+	/**
+	 * Determines if any rule supports any functions.
+	 * @param {string} property The name of the property.
+	 * @returns {boolean} `true` if any rule supports the functions, `false` if not.
+	 */
+	hasPropertyFunctions(property) {
+		return this.#rules.some(rule => rule.hasFunctions(property));
+	}
+}
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+export default {
+	meta: {
+		type: /** @type {const} */ ("problem"),
+
+		docs: {
+			description: "Enforce the use of baseline features",
+			recommended: true,
+		},
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					available: {
+						enum: ["widely", "newly"],
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+
+		defaultOptions: [
+			{
+				available: "widely",
+			},
+		],
+
+		messages: {
+			notBaselineProperty:
+				"Property '{{property}}' is not a {{availability}} available baseline feature.",
+			notBaselinePropertyValue:
+				"Value '{{value}}' of property '{{property}}' is not a {{availability}} available baseline feature.",
+			notBaselineAtRule:
+				"At-rule '@{{atRule}}' is not a {{availability}} available baseline feature.",
+			notBaselineType:
+				"Type '{{type}}' is not a {{availability}} available baseline feature.",
+		},
+	},
+
+	create(context) {
+		const availability = context.options[0].available;
+		const baselineLevel =
+			availability === "widely" ? BASELINE_HIGH : BASELINE_LOW;
+		const supportsRules = new SupportsRules();
+
+		/**
+		 * Checks a property value identifier to see if it's a baseline feature.
+		 * @param {string} property The name of the property.
+		 * @param {import("css-tree").Identifier} child The node to check.
+		 * @returns {void}
+		 */
+		function checkPropertyValueIdentifier(property, child) {
+			// named colors are always valid
+			if (namedColors.has(child.name)) {
+				return;
+			}
+			const possiblePropertyValues = propertyValues.get(property);
+
+			// if we don't know of any possible property values, just skip it
+			if (!possiblePropertyValues) {
+				return;
+			}
+
+			const propertyValueLevel = possiblePropertyValues.get(child.name);
```

Comment:
For clarity, it would be good to check if `possiblePropertyValues` has `child.name` (as it doesn't have to be the case), because if it doesn't then `propertyValueLevel` will be `undefined` and then the condition below becomes `undefined < number`.

Reply:
Ah good call. :+1: